### ISSUE TITLE
Identify compute goals/requirements

### ISSUE THREAD

**Issue by dleehr** [Duke-GCB/informatics-support-youlab#2]
Compute tasks (conversion/analysis) are currently run on laptops, and mostly within matlab. Requirements should be modest to run these on a server initially.

**Comment by dansomers** [comment]
Dan, FYI licensing for Matlab can be difficult to deal with in a shared environment. Licenses are free for Duke faculty and staff, but not for students, and licenses are tied to individuals' Mathworks accounts (also free). The OIT Site License page provides more information about what is acceptable use under Duke's licensing agreement with Mathworks. Containerizing Matlab for individual licenses might work if an arrangement cannot be (legally) made to use one person's Mathworks account on the server. Also, these accounts/licenses must be renewed annually.

Best,

-Dan

Dan Somers
Linux Systems Administrator
Center for Genomic and Computational Biology
(919) 668-3565
daniel.somers@duke.edu

**Comment by dleehr** [comment]
That's good to know. I've worked with Duke/MATLAB licensing in the past to get server license files, but that was about 5 years ago. Is there an existing matlab installation on any of our servers?

**Comment by dansomers** [comment]
There is - on compute01. It no doubt violates Duke's licensing arrangement because it uses my Mathworks account information (came up for renewal when I was a 1-man show here last Fall, and that was the quick and dirty fix). Nobody from Site License or Mathworks has come to hunt me down yet, so maybe we can get away with something similar now. I'm not fond of potentially violating the license agreement with my own account, but in certain respects it is better to have the license expiry stuff handled by someone in Informatics instead of relying on customers to track their own licenses (and notify us when they are about to expire). If we do go this route again, I'd like to clear it with Glenn Setliff, Jr at site license. I think tracking license expiry in-house may be a convincing argument for approving this use.

-Dan

Dan Somers
Linux Systems Administrator
Center for Genomic and Computational Biology
(919) 668-3565
daniel.somers@duke.edu

**Comment by dleehr** [comment]
OK, that makes sense. There's some leeway in the language though:

[Detailed License Information and Restrictions](https://software.oit.duke.edu/comp-print/software/support/matlab.php#faq-7)

> Licensed products may be installed and used in North America only by Duke faculty and academic staff on Duke-owned or leased individual, standalone computers and **network servers**, provided the programs are **not run in a cluster, grid, Web server, server farm, or other similar scheduled environment**.  Duke faculty and academic staff may install and use the programs on their personally owned computers, on campus, off campus, and via remote access.  This license includes designated computer, group license, and network concurrent options for Duke faculty and academic staff, meaning it can be installed and licensed on individual computers, in computer labs, or over a LAN using a FLEXnet server.

It sounds like it's acceptable to put your license on a server but not a cluster or behind a web service. So in this case, we might be clear, but it's still good to check.

**Comment by dansomers** [comment]
Cool - that looks like a change in licensing terms since the last time I ran up against this, and it seems clear to me from this wording that our use is acceptable. I guess other server admins here have had the same issue and Duke renegotiated that bit. Carry on!

-D

Dan Somers
Linux Systems Administrator
Center for Genomic and Computational Biology
(919) 668-3565
daniel.somers@duke.edu

**Comment by hlapp** [comment]
I agree, we should be clear so long as we don't put this on the cluster.
